bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/quickaccess/providers/CommandProvider.java

```diff
@@ -117,6 +154,33 @@ private void retrieveCommand(final String currentCommandId) {
 		}
 	}
 
+	private boolean filteredWithActivitySupport(ParameterizedCommand pc) {
+		IWorkbenchActivitySupport workbenchActivitySupport = PlatformUI.getWorkbench().getActivitySupport();
+		// fetch qualified name: plugin-is/command-name.
+		String commandId = calculateQualifiedCommandId(pc.getCommand());
+
+		if (commandId == null) {
+			return false;
+		}
+
+		IIdentifier identifier = workbenchActivitySupport.getActivityManager().getIdentifier(commandId);
+		if (!identifier.isEnabled()) {
+			return true;
+		}
+		return false;
+	}
+
+	private String calculateQualifiedCommandId(Command command) {
```

Comment:
It would be worth adding a small JavaDoc for this method and mentioning that it may return `null` (for future calls)

Reply:
Fixed.